### DISCUSSION TITLE
Api REST -> Best way for get an User id

### DISCUSSION THREAD

**Discussion by MeInTheGalaxie** [chamilo/chamilo-lms#5921] [General] [answered]
Hi,

I'm working with the Chamilo REST API, and I need to use a user's ID to be able to perform different actions.

What is the best way to get a user's ID via their "username" in your opinion?

**Comment by MeInTheGalaxie** [comment]
Good morning,
No one to give their opinion or experience on the question?

**Comment by misterakko** [comment]
Theoretically, this should give you a list of users and the id for each
curl --location 'https://YOUR_DOMAIN_NAME/main/webservices/api/v2.php?action=get_users&username=admin&api_key=YOUR_API_KEY&status=*'

On my system it returns an empty list, though.

**Reply by misterakko** [reply]
My bad... @MeInTheGalaxie, the thing needs POST, not GET. This one works:
curl -X POST --location 'https://YOUR_DOMAIN_NAME/main/webservices/api/v2.php' -H "Content-type: multipart/form-data" -F action=get_users -F username=admin -F api_key=YOUR_API_KEY -F status='*'

**Comment by MeInTheGalaxie** [comment]
Hello @misterakko, thank you for the information, indeed I use ''get_users'' also. However, this is not practical in the sense that it requires retrieving a list and working on it in order to extract the right information. It would have been nice if there was a "get_user" action that was less data intensive.

**Reply by ywarnier** [reply]
Hi @MeInTheGalaxie 

Sorry for the late answer.
Your requests sounds reasonable. In you case, what you want is a `get_user_info_from_username` where you would get something similar as the `get_users` but just for one user, based on its username as a search element, right? (just making sure that, if we create it, it matches what you're expecting)

**Reply by ywarnier** [reply, marked as answer]
I just created #6078 feature request to do that. Please continue there (rather than here) if your requirement is any different from the one described.

**Comment by MeInTheGalaxie** [comment]
@ywarnier 

Exactly, being able to retrieve the information of a single user (in particular their ID), via their unique identifier (email).

Thank you very much for taking this into account. :)

**Comment by AngelFQC** [comment]
From now, legitimate external integrations using this endpoint must do so with administrator credentials 4a119f93abbfba6fe833580f2463c8d4afa500c2
